libs/product/src/selectors/composite-product/composite-product.selectors.ts

```diff
@@ -157,9 +161,71 @@ const createCompositeProductSelectors = (): DaffCompositeProductMemoizedSelector
 		}
 	);
 
+	/**
+	 * Selector for the minimum discounted price for required items of a composite product, depending on the current selection of item options.
+	 * This would be used for a composite product that has required items without default selections.
+	 */
+	//todo use optional chaining where possible.
+	const selectCompositeProductMinDiscountedPrice = createSelector(
+		selectProductEntities,
+		selectCompositeProductAppliedOptionsEntitiesState,
+		(products, appliedOptionsEntities, props) => {
+			const product = selectProduct.projector(products, { id: props.id });
+			if(product.type !== DaffProductTypeEnum.Composite) {
+				return undefined;
+			}
+			const appliedOptions = selectCompositeProductAppliedOptions.projector(products, appliedOptionsEntities, { id: props.id });
+			return (<DaffCompositeProduct>product).items.reduce((acc, item) => daffAdd(
+				acc, 
+				appliedOptions[item.id] ? 
+					daffMultiply(
+						daffSubtract(appliedOptions[item.id].price, appliedOptions[item.id].discount && appliedOptions[item.id].discount.amount), 
```

Comment:
I'd actually rather change the daff arithmetic functions to be able to handle null values instead of adding more logic here. I'll probably make a PR for this tomorrow.

Reply:
okay sounds good